### ISSUE TITLE
formater: Add `getFormatterList` function

### ISSUE THREAD

**Issue by azu** [textlint/textlint#399]
[textlint-formatter](https://github.com/textlint/textlint/tree/master/packages/textlint-formatter "textlint-formatter") and [textlint-fixer-formatter](https://github.com/textlint/textlint/tree/master/packages/textlint-fixer-formatter "textlint-fixer-formatter") does not export own formatter list like `"stylish"`, `"compats"` etc...

It need to resolve #85

- [show available formatter in help · Issue #85 · textlint/textlint](https://github.com/textlint/textlint/issues/85 "show available formatter in help · Issue #85 · textlint/textlint")

## Tasks

- [x] Add `getFormatterList` function to [textlint-formatter](https://github.com/textlint/textlint/tree/master/packages/textlint-formatter "textlint-formatter")  #415 
- [x] Add `getFormatterList` function to [textlint-fixer-formatter](https://github.com/textlint/textlint/tree/master/packages/textlint-fixer-formatter "textlint-fixer-formatter") #412 

`getFormatterList()` returns an array of `{ name: "formatter name"}` object.

For example, [textlint-formatter](https://github.com/textlint/textlint/tree/master/packages/textlint-formatter "textlint-formatter")'s `getFormatterList()`  returns following array of object:

```js
[{
  name: "stylish"
}, {
  name: "compact"
}]
```

- ... [textlint-formatter/lib/formatters/](https://github.com/textlint/textlint/tree/master/packages/textlint-formatter/lib/formatters "textlint-formatter/lib/formatters/") file names ...

Probably, [fs.readdirSync](https://nodejs.org/api/fs.html#fs_fs_readdirsync_path_options "fs.readdirSync") help this implementation.

**Comment by azu** [comment]
I've edit `getFormatterList()` result value.

```diff
- return an array of formatter names
+ return an array of `{ name: "formatter name"}` object
```

It aim to get extensibility.

**Comment by azu** [comment]
👏 Closed by @0x6b PR #415 #412
